### PR TITLE
HIVE-24541: add config option for default storage handler class

### PR DESCRIPTION
**Pull request by marton-bod**
### What changes were proposed in this pull request?
Add new config option to set a default storage handler class. If this config is set to a valid, existing class, the storage handler class won't have to be explicitly provided in the create table statement with the `STORED BY `clause, but will be 'inserted' under the hood.

For example, if `hive.default.storage.handler.class=org.apache.iceberg.mr.hive.HiveIcebergStorageHandler` is set,
then the command `CREATE TABLE abc (a int, b string) LOCATION ...` will be equivalent to 
`CREATE TABLE abc (a int, b string) STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' LOCATION ...`

### Why are the changes needed?
In order to use existing create table commands for new table formats, such as Iceberg, without modification.

### Does this PR introduce _any_ user-facing change?
new config option settable via beeline/hive-site

### How was this patch tested?
q tests + manually with local run

### CODE REVIEW

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java [hunk 1]
```diff
@@ -1991,6 +1991,10 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "Default file format for CREATE TABLE statement applied to managed tables only. External tables will be \n" +
         "created with format specified by hive.default.fileformat. Leaving this null will result in using hive.default.fileformat \n" +
         "for all tables."),
+    HIVEDEFAULTSTORAGEHANDLER("hive.default.storage.handler.class", "",
```
Comment (pvary) [inline common/src/java/org/apache/hadoop/hive/conf/HiveConf.java:1994]: Could be please use underscores to separate names? I understand that the ones before and after are not using them, but currently the underscore is the "standard"

File: ql/src/java/org/apache/hadoop/hive/ql/parse/StorageFormat.java [hunk 2]
```diff
@@ -69,6 +69,12 @@ public boolean fillStorageFormat(ASTNode child) throws SemanticException {
       }
       break;
     case HiveParser.TOK_FILEFORMAT_GENERIC:
+      if (storageHandler != null) {
+        // Under normal circumstances, we should not get here (since STORED BY and STORED AS are incompatible within the
+        // same command). Only scenario we can end up here is if a default storage handler class is set in the config.
+        // In this case, we opt to ignore the STORED AS clause.
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/StorageFormat.java:75]: Maybe log?

Comment (marton-bod) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/StorageFormat.java:80]: sure

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java (hunk 1 above)
Comment (marton-bod) [reply common/src/java/org/apache/hadoop/hive/conf/HiveConf.java]: sure

**Review by pvary — APPROVED** [review]
+1 pending tests

**Pull request merged**